### DISCUSSION TITLE
Vmess 中入站连接为 zero 加密方式时，似乎不会进入 InsecureEncryption 的处理逻辑，这样合理吗？

### DISCUSSION THREAD

**Discussion by IceCodeNew** [v2fly/v2ray-core#736] [General]
看了下源代码，不确定是不是有意这么设计的。

https://github.com/v2fly/v2ray-core/blob/733c5dea885969d2d81e01cd921fb1cc7ea9405c/proxy/vmess/inbound/inbound.go#L215-L217
https://github.com/v2fly/v2ray-core/blob/1368e926057a0d1d2efe4322926aad1af6ea8aee/infra/conf/vmess.go#L34-L35

**Comment by xiaokangwang** [comment]
https://github.com/v2fly/v2ray-core/blob/1368e926057a0d1d2efe4322926aad1af6ea8aee/proxy/vmess/outbound/outbound.go#L113-L117
实际上对于 VMess 服务器端来说并不存在Zero这个加密方式，这个是在客户端虚构出来的。对于服务器来说是None加上几个VMess特殊的包头选项。

正因如此这个才是客户端修改，完全不需要服务器端的更新就能支持。

**Reply by IceCodeNew** [reply]
我明白了，所以其实是会进入 InsecureEncryption 的处理逻辑，感谢回答！
